File: apps/photog/lib/photog/image/image.ex
defmodule Photog.Image do
  alias Photog.Api.Image

  def masters_directory do
    Path.expand("../../../priv/static/media/images", __DIR__)
  end

  def thumbnails_directory do
    Path.expand("../../../priv/static/media/thumbnails", __DIR__)
  end

  def master_file_path(%Image{} = image) do
    masters_directory()
    |> Path.join(image.master_path)
  end
end

File: apps/photog/assets/js/image.js
const joinUrlParts = urlParts => urlParts.filter(part => !!part).join('/');

export const thumbnailUrlFor = (path, prefix = '') =>
    path ? joinUrlParts(['/media/thumbnails', prefix, encodeURI(path)]) : '';

export const getMasterUrl = (image, prefix = '') =>
    joinUrlParts(['/media/images', prefix, encodeURI(image.master_path)]);
